01_financial-well-being/income_metrics_county.qmd

```diff
@@ -0,0 +1,735 @@
+  + 2 Black/African American/Negro
+  + 3 American Indian or Alaska Native 
+  + 4 Chinese 
+  + 5 Japanese
+  + 6 Other Asian or Pacific Islander 
+  + 7 Other race 
+  + 8 Two major races 
+  + 9 Three or more major races
+
+* IPUMS documented values for HISPAN: 
+  + 0 Not Hispanic 
+  + 1 Mexican 
+  + 2 Puerto Rican 
+  + 3 Cuban 
+  + 4 Other 
+  + 9 Not Reported 
+
+Check that group quarters fall only into household categories.
+```{r}
+
+acs_clean %>% 
+  count(race) %>% 
+  ggplot(mapping = aes(x = race, y = n)) +
+  geom_col() +
+  theme_minimal() +
+  ggtitle("Race, All Records") +
+ geom_text(mapping = aes(label = n), vjust = -1) +    
+  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
+  labs(x = "Race",
+       y = NULL)
+
+acs_clean %>% 
+  count(hispan) %>% 
+  ggplot(mapping = aes(x = hispan, y = n)) +
+  geom_col() +
+  theme_minimal() +
+  ggtitle("Hispan, All Records") +
+ geom_text(mapping = aes(label = n), vjust = -1) +    
+  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
+  labs(x = "Hispan",
+       y = NULL)
+
+```
+
+Create the race/ethnicity variable. For other race/ethnicity categories not coded as Hispanic, Hispan must be equal to "Not Hispanic". 
+
+Rename the values from number category to race label: 
+  +Race is 2 and Hispan is 0 = "Black, Non-Hispanic"
+  +Hispan is >=1 and <=4 = "Hispanic"
+  +Race is >=3 and <=9 and Hispan is 0  = "Other Races and Ethnicities" 
+  +Race is 4 and Hispan is 0 = "White, Non-Hispanic" 
+```{r}
+
+acs_clean <- acs_clean %>%
+  mutate(subgroup = case_when((hispan == "Not Hispanic" & race == "White") ~ "White, Non-Hispanic",
+                              (hispan == "Not Hispanic" & race == "Black/African American") ~ "Black, Non-Hispanic",
+                              (hispan == "Not Hispanic" & !race %in% c("Black/African American", "White")) ~ "Other Races and Ethnicities",
+                               hispan != "Not Hispanic" ~ "Hispanic"
+  ))
+
+```
+
+Look at subgroup distribution.
+
+```{r}
+
+acs_clean %>% 
+  count(subgroup) %>% 
+  ggplot(mapping = aes(x = factor(subgroup), y = n)) +
+  geom_col() +
+  theme_minimal() +
+  ggtitle("Subgroup, All Records") +
+ geom_text(mapping = aes(label = n), vjust = -1) +    
+  scale_y_continuous(expand = expansion(mult = c(0, 0.1))) +
+  labs(x = "Subgroup",
+       y = NULL)
+```
+
+## Merge Crosswalk 
+
+Create a variable in the ACS data indicating whether the survey is pre or post-2022. This will be used to join on the correct crosswalk information.
+```{r}
+acs_clean <- acs_clean %>% 
+  mutate(crosswalk_period = ifelse(year < 2022, "pre-2022", "2022"))
+
+```
+
+Check that the assignment worked. All years prior to 2022 should be designated "pre-2022".
+```{r}
+acs_clean %>% 
+  group_by(crosswalk_period, year) %>% 
+  count()
+```
+
+Join the cleaned ACS data onto the crosswalk.
+```{r}
+
+acs_clean <- left_join(acs_clean, puma_county_crosswalk, by=c("crosswalk_period", "statefip","puma"))
```

Comment:
throws Warning message:
In left_join(acs_clean, puma_county_crosswalk, by = c("crosswalk_period",  :
  Detected an unexpected many-to-many relationship between `x` and `y`.
ℹ Row 1 of `x` matches multiple rows in `y`.
ℹ Row 4751 of `y` matches multiple rows in `x`.
ℹ If a many-to-many relationship is expected, set `relationship = "many-to-many"` to silence this warning.

Reply:
There is a person who has multiple matches in "crosswalk_period", "statefip","puma"... well more than one, it looked like the population is doubling. from 41 million to 85 million. Is this error present when you run this line?